src/useObserver.ts

```diff
@@ -63,3 +65,64 @@ export function useObserver<T>(
     }
     return rendering
 }
+
+/**
+ * Reactions created by components that have yet to be committed.
+ */
+const uncommittedReactions: Reaction[] = []
+
+/**
+ * Latest 'uncommitted reactions' cleanup timer handle
+ */
+let reactionCleanupHandle: number | undefined
+
+function scheduleCleanupOfReactionIfNotCommitted(reaction: Reaction) {
+    uncommittedReactions.push(reaction)
+    if (reactionCleanupHandle) {
+        window.clearTimeout(reactionCleanupHandle)
+    }
+    // We currently have no cleanup timer running; schedule one
+    reactionCleanupHandle = window.setTimeout(cleanUncommittedReactions, 100)
+}
+
+function recordReactionAsCommitted(reaction: Reaction) {
+    // It would be more efficient if we could use a Set instead of an Array,
+    // but mobx-react-lite currently supports MobX 4 and ES5, so we can't assume
+    // the presence of Set.
+    const i = uncommittedReactions.indexOf(reaction)
```

Comment:
Nit: reactions have a unique id, so you could use those in an object index for quicker cleanup. But I think you can safely assume Set / Map, as React 16 itself needs those to be available / polyfilled :)